trade-finance-manager-api/src/v1/helpers/amendment.helpers.js

```diff
@@ -515,13 +514,34 @@ const addLatestAmendmentDates = async (tfmObject, latestCoverEndDateResponse, la
 
   const coverEndDate = latestCoverEndDateResponse?.coverEndDate;
   const amendmentExposurePeriodInMonths = coverEndDate ? await calculateAmendmentDateTenor(coverEndDate, existingFacility) : undefined;
-  const facilityEndDate = latestFacilityEndDateDataResponse?.facilityEndDate;
 
   return {
     ...tfmObject,
     coverEndDate,
     amendmentExposurePeriodInMonths,
-    facilityEndDate,
+  };
+};
+
+/**
+ * Populates the tfmObject with facility end date values
+ * @param {import('@ukef/dtfs2-common').FacilityAmendmentTfmObject} tfmObject
+ * @param {{facilityEndDate?: string, bankReviewDate?: string, isUsingFacilityEndDate?: boolean }} latestFacilityEndDateDataResponse
+ * @param {string} facilityId
+ * @returns {Promise<import('@ukef/dtfs2-common').FacilityAmendmentTfmObject>}
+ */
+const addLatestAmendmentFacilityEndDate = async (tfmObject, latestFacilityEndDateDataResponse, facilityId) => {
+  const existingFacility = await api.findOneFacility(facilityId);
```

Comment:
Wait a minute why is existingFacility here at all? Is it used?

Reply:
A really great spot, that isn't needed at all😅 (copy and paste hangover from the cover end date function but there it is actually used)

Reply:
Have removed